pandas/io/parsers.py

```diff
@@ -1550,11 +1569,22 @@ def close(self):
 
     def _set_noconvert_columns(self):
         names = self.orig_names
-        usecols = self.usecols
+        if self.usecols_dtype == 'integer':
+            # A set of integers will be converted to a list in
+            # the correct order every single time.
+            usecols = list(self.usecols)
```

Comment:
so this is *very* confusing here. You are now converting to a list. So ``usecols`` ends up as a list, None or a callable? when / how does it end up as a set?